source/slang/slang-check-decl.cpp

```diff
@@ -7119,6 +7119,99 @@ namespace Slang
 
         return args;
     }
+    
+    SubstitutionSet makeSubstitutionFromIncompleteSet(
```

Comment:
You can call `SemanticsVisitor::inferGenericArguments`, which implement the same logic. Let's try not duplicate the logic here.

Reply:
I think this specific function is different from `inferGenericArguments`. We want to use it for `specializeGeneric` which operates on struct types as well as callables, where we are provided specific argument substitutions for the generic parameters directly (not the arg types of the callables).

I plan on providing a separate function `FunctionReflection::specializeWithArgTypes(...)` that will invoke `inferGenericArguments` directly.

Alternatively, I can change up `inferGenericArguments` to also support struct types. With generic interfaces, we probably need to enforce constraints across the arguments of the same generic.

Reply:
`inferGenericArguments` is not only for function calls but for types as well.

Basically if you have `S<T:IFoo, int N =  4>` and you pass in `S<FooImpl>`, `inferGenericArguments` will give you `S<FooImpl, 4, FooImpl : IFoo>`.

Reply:
I guess for here, you wan't to reuse or factor out `checkGenericAppWithCheckedArgs`.

Reply:
Rerote this method with `checkGenericAppWithCheckedArgs` + added support for bool type args.

I think this is a good suggestion, since the new version should work seamlessly with variadics.

Reply:
We can delete this now?